roles/openshift_health_checker/library/search_journalctl.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python
+"""Interface to journalctl"""
+
+from time import time
+import json
+import re
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+class InvalidMatcherRegexp(Exception):
+    """Exception class for invalid matcher regexp."""
+    pass
+
+
+class InvalidMatcher(Exception):
+    """Exception class for missing or invalid matcher fields."""
+    def __init__(self, message, missing_fields=None):
+        super(InvalidMatcher, self).__init__(message)
+        self.missing_fields = missing_fields
+
+
+class InvalidLogEntry(Exception):
+    """Exception class for invalid / non-json log entries."""
+    pass
+
+
+class LogInputSubprocessError(Exception):
+    """Exception class for errors that occur while executing a subprocess"""
+    pass
+
+
+def main():
+    """Scan a given list of matchers for journalctl messages containing given patterns."""
+    module = AnsibleModule(
+        argument_spec=dict(
+            log_count_limit=dict(type="int", default=500),
+            log_matchers=dict(type="list", required=True),
```

Comment:
Perhaps we should document better what is understood to be a "matcher"?
Reading through the code I think it is a dict with 3 keys 'start_regexp', 'regexp' and 'unit'?

I've always wondered if we can represent more complex structures in the `argument_spec`...